src/main/java/org/osiam/resources/scim/UpdateUser.java

```diff
@@ -0,0 +1,858 @@
+            Entitlement deleteEntitlement = new Entitlement.Builder()
+                    .setValue(entitlement.getValue())
+                    .setType(entitlement.getType())
+                    .setOperation(DELETE)
+                    .build();
+            entitlements.add(deleteEntitlement);
+            return this;
+        }
+
+        /**
+         * adds or updates a entitlement to an existing user if the .getValue() already exists a update will be done. If
+         * not a new one will be added
+         *
+         * @param entitlement new entitlement
+         * @return The builder itself
+         */
+        public Builder addEntitlement(Entitlement entitlement) {
+            entitlements.add(entitlement);
+            return this;
+        }
+
+        /**
+         * updates the old Entitlement with the new one
+         *
+         * @param oldEntitlement to be replaced
+         * @param newEntitlement new Entitlement
+         * @return The builder itself
+         */
+        public Builder updateEntitlement(Entitlement oldEntitlement, Entitlement newEntitlement) {
+            deleteEntitlement(oldEntitlement);
+            addEntitlement(newEntitlement);
+            return this;
+        }
+
+        /**
+         * updates the active status of a existing User to the given value
+         *
+         * @param active new active status
+         * @return The builder itself
+         */
+        public Builder updateActive(boolean active) {
+            this.active = active;
+            return this;
+        }
+
+        /**
+         * deletes the given extension of a existing user
+         *
+         * @param urn the id of the extension to be deleted
+         * @return The builder itself
+         */
+        public Builder deleteExtension(String urn) {
+            deleteFields.add(urn);
+            return this;
+        }
+
+        /**
+         * deletes the given extension field of a existing user
+         *
+         * @param urn       the id of the extension to be deleted
+         * @param fieldName the fieldName of a the extension to be deleted
+         * @return The builder itself
+         */
+        public Builder deleteExtensionField(String urn, String fieldName) {
+            deleteFields.add(urn + "." + fieldName);
+            return this;
+        }
+
+        /**
+         * updates the given fields in the extension. If the User doesn't have the given extension fields, they will be
+         * added.
+         *
+         * @param extension extension with all fields that need to be updated or added
+         * @return The builder itself
+         */
+        public Builder updateExtension(Extension extension) {
+            extensions.add(extension);
+            return this;
+        }
+
+        /**
+         * constructs a UpdateUser with the given values
+         *
+         * @return a valid {@link UpdateUser}
+         */
+        public UpdateUser build() {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The method build() has an NPath complexity of 12582912](https://www.codacy.com/app/OSIAM/osiam/file/1979525074/issues/source?bid=3080997&fileBranchId=3117040#l780)